pkg/core/native/native_neo.go

```diff
@@ -727,13 +727,25 @@ func (n *NEO) ComputeCommitteeMembers(bc blockchainer.Blockchainer, d dao.DAO) (
 }
 
 func (n *NEO) getNextBlockValidators(ic *interop.Context, _ []stackitem.Item) stackitem.Item {
-	result := n.GetNextBlockValidatorsInternal()
+	result, err := n.GetNextBlockValidatorsInternal(ic.Chain, ic.DAO)
+	if err != nil {
+		panic(err)
+	}
 	return pubsToArray(result)
 }
 
-// GetNextBlockValidatorsInternal returns next block validators.
-func (n *NEO) GetNextBlockValidatorsInternal() keys.PublicKeys {
-	return n.nextValidators.Load().(keys.PublicKeys).Copy()
+// GetNextBlockValidatorsInternal returns next block validators using cached value.
+func (n *NEO) GetNextBlockValidatorsInternal(bc blockchainer.Blockchainer, d dao.DAO) (keys.PublicKeys, error) {
+	if n.nextValidators.Load().(keys.PublicKeys) == nil {
+		committee, err := n.ComputeCommitteeMembers(bc, d)
```

Comment:
`ic.DAO.GetStorageItem(n.ContractID, prefixCommittee)`

It's explicitly stored in the DB, you should just get it here instead of computing.

Reply:
And I think it's way easier to just add some InitializeInternal methods to all contracts and invoke them from `(bc *Blockchain) init()`. These methods would just get everything needed from the DB and you could rely on this initialization (saving lots of potential runtime errors).